packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-token.ts

```diff
@@ -50,24 +48,21 @@ defineProperty(Element.prototype, KEY__SHADOW_TOKEN, {
     configurable: true,
 });
 
+function recursivelySetShadowResolver(node: Node, fn: any) {
+    (node as any)[KEY__SHADOW_RESOLVER] = fn;
+
+    const { childNodes } = node;
```

Comment:
should we use [childNodesGetter from env](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/synthetic-shadow/src/env/node.ts#L100)?

Reply:
Done.

I also decided to cache the `length` in the for-loop, because even though I said we don't need to, I don't trust that V8 optimized it for HTMLCollection (non-Array), and I'm too lazy to test. 😛

Reply:
OK, I got un-lazy and benchmarked it. There might be like a <1% improvement from caching the `length`, but it's not super clear.

![Remote Desktop Picture August 8, 2022 at 10 30 33 AM PDT](https://user-images.githubusercontent.com/283842/183478246-deeaaa0a-2e88-4b15-84b8-a080a246ab85.png)